src/ripple/json/impl/json_value.cpp

```diff
@@ -1068,6 +1068,24 @@ Value::get ( std::string const& key,
 }
 
 Value
+Value::removeElement ( std::uint32_t index )
+{
+    JSON_ASSERT (type_ == arrayValue || index < value_.map_->size ());
```

Comment:
@rec Yes, the assert should use &&

@vinniefalco removeElement can be dangerous if you are holding a pointer/ref to an element. The existing removeMember function already presents this danger. That is why I added a warning in the declarations.